jupyterhub/orm.py

```diff
@@ -230,7 +231,12 @@ class Spawner(Base):
     user_id = Column(Integer, ForeignKey('users.id', ondelete='CASCADE'))
 
     server_id = Column(Integer, ForeignKey('servers.id', ondelete='SET NULL'))
-    server = relationship(Server, backref='spawner', cascade="all")
+    server = relationship(
```

Comment:
I was just going to point out that this is similar to how https://github.com/jupyterhub/jupyterhub/blob/84acdd5a7ffb3d79ca04aa86ffcea7e067cce6ee/jupyterhub/orm.py#L175 works which implicitly creates a `Spawner.user` attribute, the only difference being `uselist=False` and `single_parent=True`.

I've worked on another project that also used `uselist=False` in a couple of places, e.g.:

https://github.com/openstack/nova/blob/b6e353a2c84d87bbb1353ff6602f9f13dc62a2fd/nova/db/sqlalchemy/api_models.py#L151

And that's pretty similar to how you're using it here (a `cell_mapping` can have many `instance_mappings` but an `instance_mapping` can only ever be in one `cell_mapping`).